pandas/core/aggregation.py

```diff
@@ -5,14 +5,72 @@
 
 from collections import defaultdict
 from functools import partial
-from typing import Any, DefaultDict, List, Sequence, Tuple
+from typing import Any, DefaultDict, List, Optional, Sequence, Tuple, Union
 
 from pandas.core.dtypes.common import is_dict_like, is_list_like
+from pandas.core.dtypes.generic import ABCDataFrame
 
+from pandas.core.base import SpecificationError
 import pandas.core.common as com
 from pandas.core.indexes.api import Index
 
 
+def reconstruct_func(
+    func: Optional[Union[List[Any], dict]], *args, **kwargs: dict
+) -> Tuple[bool, Any, Optional[List[str]], Optional[List[int]]]:
```

Comment:
```suggestion
) -> Tuple[bool, Optional[Callable], Optional[List[Label]], Optional[List[int]]]:
```

Reply:
thanks!

Reply:
emm, seem this isn't correct, i slightly changed a bit, advice are wecome